doc/upstream_release_process.md

```diff
@@ -1,152 +1,166 @@
-with 'lp-project-upload' from 'lptools' package:
-
-
-    $ lp-project-upload cloud-init 17.2 cloud-init-17.2.tar.gz 17.3 changelog-file releasenotes-file
-
+Go to https://launchpad.net/cloud-init click the milestone that we're
+releasing.
+
+Hit 'Create release' (Under Milestone information in middle of page)
+
+Fill in details:
+* **Don't** keep milestone active
+* Specify current date for 'Date released'
+* Copy the release notes from the bug (minus the changelog) into the 'Release notes' section
+* Copy the changelog from the bug into the 'Changelog' section.
+
+Once we have created our release, we should create the milestone for the next release.
+* From https://launchpad.net/cloud-init , under `Series and milestones` (middle of page), click the `trunk` series. If you see no `trunk` series, then click `View full history` and find it there
+* Under the `Milestones and releases` section, click the `Create milestone` button
+* Set Name to `<next scheduled release>`. For example, if current release is '21.3', Name should be '21.4'
+* If we have a scheduled date for the next release, set the date, otherwise leave it blank
+* Leave everything else blank, and click `Create Milestone`
+
+## Create release in github
+* Visit https://github.com/canonical/cloud-init/releases
+* Click 'Draft new release'
+* In the 'Choose a tag' dropdown, select tag pushed earlier
+* Set `<version>` as Release title
+* Set `'Release <version>'` as the description
+* Click 'Publish release'
 
 ## Close bugs.
-Any bugs that were listed should be marked as 'fix-released' now.
-There is a tool in
-[uss-tableflip](https://github.com/CanonicalLtd/uss-tableflip) called
-lp-bugs-released that makes this sane.
-
-    # git log <last-release>..<this-release>
-    $ git log 17.1..17.2 | grep "^[ ]*LP:" | sort -u
-    $ ./lp-bugs-released <project> <version> <bug list here>
-
-Basically copy the Release Notes and the Changelog into an email to
-Cloud-init Mailing List <cloud-init@lists.launchpad.net>
+Any Launchdpad bugs that were listed in the git commit messages from this release should be marked as 'fix-released' now.
 
-Example Emails from the past at
-
- * [17.1](https://lists.launchpad.net/cloud-init/msg00106.html)
-
-Please sign the email.
-
-
-## Upload to Ubuntu devel release.
-Follow the Ubuntu release process doc [ubuntu-release-process](https://gist.github.com/smoser/6391b854e6a80475aac473bba4ef0310#file-ubuntu-release-process-md)
+There is a tool in
+[uss-tableflip](https://github.com/canonical/uss-tableflip/) called
+lp-bugs-released that makes this sane:
+```bash
+$ ./lp-bugs-released <project> <version> <space separated list of bugs>
+```
+Example:
+```bash
+$ ./lp-bugs-release cloud-init 21.3 1867532 1911680 1925395 1931392 1931577 1932048 1940233 1940235 1940839
+```
+
+To get the list of bugs that have been fixed this release, use:
+```bash
+$ git log <previous_version>..<version> | grep "^[ ]*LP:" | sort -u | awk -F 'LP: #' '{printf $2 " "}'
```

Comment:
Maybe move this up just above the lp-bugs-released <project> line as it's pre-requisite